### DISCUSSION TITLE
Sync OProperties in OClass and corresponded records?

### DISCUSSION THREAD

**Discussion by dimitrievgs** [orientechnologies/orientdb#9652] [Q&A]
Hello, is there any possibility to sync OProperties in OClass and corresponded records?
I previously asked about it in https://github.com/orientechnologies/orientdb/issues/9632 but nobody answered.

It's just question about possibility to use orientdb for such tasks. So it will be just like tables in relational databases: I change tables, and corresponded entitites are changed too (right after it).

So to have this sync at one hand and have as many bonds (OEdges) with other records and of as much types as one wants.

And probably is it possible to add custom fields to the record: so it has a group of synchronizable properties (from OClass) and local (individual) properties in each record?

**Comment by luigidellaquila** [comment]
Hi @dimitrievgs 

I just answered on the other thread. Unfortunately OrientDB does not implement such behaviour, so you have to do it with an UPDATE

Thanks

Luigi

**Reply by dimitrievgs** [reply]
Actually it makes no difference whether or not I use a command 
UPDATE Foo SET theNewProp = theOldProp

Now I change the code to run all commands:
"UPDATE `" + oClass.getName() + "` SET `" + newPropertyName + "` = `" + oldPropertyName + "`";
"ALTER PROPERTY `" + oClass.getName() + "`.`" + oldPropertyName + "` NAME `" + newPropertyName + "`";
"UPDATE `" + oClass.getName() + "` REMOVE `" + oldPropertyName + "`";

And it just change the class but not the corresponded records. The value is not transferred to another (new) field until I reboot the program which usess the orientdb (probably it updates corresponded records only during database closing or opening, I don't know). But to make it usable I need to changes to take place in the same program run.
The same behaviour both in versions 3.0.37 and 3.1.11.

Is it possible to make the orientdb transfer data right after changing the OClass? "UPDATE Foo SET theNewProp = theOldProp" doesn't help.

**Comment by PhantomYdn** [comment]
Let me also answer to the second part of the question: 

```
And probably is it possible to add custom fields to the record: so it has a group of synchronizable properties (from OClass) and local (individual) properties in each record?
```

Yes - it's possible. You can have on any record as many extra fields as you want. But be cautious when you might want to add new property: if its name will be the same as some custom field - you will have troubles.
